### PR TITLE
Adds Surgery Skillchips

### PR DESCRIPTION
**Pull request by maxspells**
## About The Pull Request
Adds a “Surgery skillchip” which adds a surgeon trait. Players without the trait will have a base 25% chance to fail surgery steps no matter how good the conditions get, this number will shoot up to 60% if you have chunkyfingers, provided you are not an abductor. Medical Doctors and CMOs will start with one, and the Roboticist skillchip will come with the trait as well. The trait has also been added to relevant places such as abductors, medical borgs, and the Rod of Asclepius.

## Why It's Good for the Game
I detailed my reasoning for this in a design doc you can find [here](https://hackmd.io/fg55of5WSSSLOojGjqkxNQ?view).

Medbay, more specifically the Medical Doctor, has no barrier for entry in its current state for people outside of the department. Surgical tools are easily obtained, to the point where it is simple for chemists or parameds to just waltz into med storage, print tools and instantly be able to play as a MD with better access to chem or other departments.

## Changelog

:cl:
add: Surgical skillchips
add: Surgery traits to relevant places
balance: Players without the trait will always have a chance to fail surgery steps, no matter how good the conditions are.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/silicon/robot/robot_model.dm [hunk 1]
```diff
@@ -620,6 +620,7 @@
 	cyborg_base_icon = "medical"
 	model_select_icon = "medical"
 	model_traits = list(TRAIT_PUSHIMMUNE)
+	model_traits = list(TRAIT_SURGEON)
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/silicon/robot/robot_model.dm:623]:
```suggestion
	model_traits = list(TRAIT_PUSHIMMUNE, TRAIT_SURGEON)
```

**Comment by WarlockD** [comment]
I would say this might get not merged, but its been years since chems and surgery has been redone.  Is a failure mechanic possible for non doctors atleast on major surgery?  I don't remember the conversations when the skill system for surgery was pulled.

**Comment by Cheshify** [comment]
I see what people mean when they say "coders hate fun" now.

**Comment by Cheshify** [comment]
I don't think we should disincentivize play we don't want to see, and instead incentivize play we want to see. This is just a nerf to people doing gimmicks, antag gimmicks, doing surgery in emergencies, and shifts without doctors. If there NEEDS to be a skillchip for doctors, why not have it make surgery faster for them?

**Comment by WarlockD** [comment]
> I see what people mean when they say "coders hate fun" now.

Nah, we live of the tears of other coders and memes.  I mean, I am not agents failure mechanics, I just know that the whole reason the surgery system was revamped was because people ignored medical because of that.  We couldn't even think of removing the clone vat till atleast surgery was good enough people wanted to play.

**Comment by Cheshify** [comment]
> Nah, we live of the tears of other coders and memes. I mean, I am not agents failure mechanics, I just know that the whole reason the surgery system was revamped was because people ignored medical because of that. We couldn't even think of removing the clone vat till atleast surgery was good enough people wanted to play.

deleted that for a reason, it was salty and mean, and I don't really agree with it

**Comment by WarlockD** [comment]
> I don't think we should disincentivize play we don't want to see, and instead incentivize play we want to see. This is just a nerf to people doing gimmicks, antag gimmicks, doing surgery in emergencies, and shifts without doctors. If there NEEDS to be a skillchip for doctors, why not have it make surgery faster for them?

Mechanic isn't as fun.  I can say that with both experience and the yelling at the end.   We had a skill pr similar to the cleaning/mining skill mechanic for doctors.  End up doctors would grind away at corpses or near corpses to level it up to become gods.  The second it comes in, it will be mandatory for any doctors to have it.  Also unless somone has like a box of them, like a big box not like engineers, no one wil want to play doctor if they spawned in as something else.

Don't worry about being salty, as long as we know.  There are more meme prs than I know and its all turtles down here.

**Comment by WarlockD** [comment]
I mean think of it like this. 

1.  If you spawn a doctor, this whole pr doesn't affect you have to yank one out for a traitor requirement.  
2. You don't spawn a doctor, but the station needs them.   No skill chip means no doctor because no one wants to play with a faiure rate that you cannot control
3. Speed doesn't matter.  Its all about if you get to them in that first 5m before you have to do heart surgery.  At that point its all about if you pre-pared a table for those situations before hand, that goes a farther way than even a 25% boost

Humm.  What instead of a failure, you can do more than one surgery at a time?  They pulled that "feature" out a while back so borg surgery is a crawl now because of it.

**Comment by WarlockD** [comment]
I also want to incentivize ghetto surgery.   We need some getto implants, even if they don't really work the way as intended hah.

**Comment by SkeletalElite** [comment]
What if you give people with the skill chip an interface or something that tells them the next surgery step without an operating computer?

That would be more like the atmos tech to engineer comparison. An atmos tech can still interact with wires just as well as an engineer but they have to be more knowledgeable about wires since the game doesn't just tell them what wires to use.

**Comment by timothymtorres** [comment]
One of the things that makes SS13 great is the fact that ANY player can do any job if they have the OOC knowledge to perform it.   Medbay already requires special ID card access that serves as a barrier from preventing people from getting in.  Every department has to deal with the greytide.

I'm not a fan of locking cargo/security/hydro/engineering/science/chemistry behind skillchips.  It's better to make the content more complicated or hard (like Colby Chem) than just outright denying people the ability to perform tasks.  Granted, your changes only reduces the chances of a successful surgery step, but I think this is still headed in the wrong direction.

-1 from me.

**Comment by ATH1909** [comment]
fucks over ghetto surgery and free golem AI construction, -1 from me

how does this interact with that prog traitor brainwashing surgery objective?

I'm not really convinced by your design doc. I don't think that chemists step on the toes of doctors any more than, say, doctors step on the toes of chemists. yeah, randos can walk in and help with surgery, but that's a GOOD thing. those are people who are, out of the kindness of their heart, taking time out of their shift purely to help other people get back into the game. why are you saying "no, fuck you" to them? it's not like they're not going to continue to try to do surgery, they'll just do it with a frustrating amount of rng.

if you want to add a surgery skill chip, just make having it multiply surgery times by 0.8x or something and not having it use the current times (and/or, as someone else suggested earlier, let it let you perform surgery on multiple people at once).

**Comment by WarlockD** [comment]
> That would be more like the atmos tech to engineer comparison. An atmos tech can still interact with wires just as well as an engineer but they have to be more knowledgeable about wires since the game doesn't just tell them what wires to use.
Yea!  You can even go further by saying you have access to more surgerys at the start.  Mabye the ablity to do some surgerys with the clothes on since you cloth's fu?

**Comment by ATH1909** [comment]
> > That would be more like the atmos tech to engineer comparison. An atmos tech can still interact with wires just as well as an engineer but they have to be more knowledgeable about wires since the game doesn't just tell them what wires to use.
> > Yea!  You can even go further by saying you have access to more surgerys at the start.  Mabye the ablity to do some surgerys with the clothes on since you cloth's fu?

locking off surgeries entirely behind the skill chip is even worse, that's just denying everyone else access to them at all

**Comment by WarlockD** [comment]
Lord no, not locking.  Just like having access to like a level of surgery's eairlyer.  Like when you have a nuke event, the doctors don't have to worry to much about getting the upgraded stuff.

**Comment by maxspells** [comment]
> Also unless somone has like a box of them, like a big box not like engineers, no one wil want to play doctor if they spawned in as something else.

I'm super open to the idea of the CMO office having extra skillchips roundstart.

As for the speed nerf, I thoroughly playtested this with ghetto surgery and on average I would fail a step once per surgery. At worse that adds a few seconds or so depending on how bad the conditions are, so it is a very minor inconvenience to people who still want or need to do surgery without the chip, and is actually much less of a nerf than if I were to increase the surgery times instead.

**Comment by WarlockD** [comment]
> As for the fail% nerf, I thoroughly playtested this with ghetto surgery and on average I would fail a step once per surgery. At worse that adds a few seconds or so depending on how bad the conditions are, so it is a very minor inconvenience to people who still want or need to do surgery without the chip, and is actually much less of a nerf than if I were to increase the surgery speeds instead.

... I just want to see this pr test merge to watch those tear's.  Fair warning:)

**Comment by MagmaExploiter** [comment]
Most existing role-specific skillchips (engineer and roboticist in particular) exist to give a bonus to that role's core tasks, rather than just penalizing people for attempting it without the skillchip, and I think that would be a better direction for this.

If you're dead set on a surgery fail chance without the skillchip, I'd suggest giving ways around it with extra prep. Surgical sterilizers in particular come to mind - I remember reading they give a success rate bonus on top of the usual speed boost, but I've never seen it in action myself. Upping the success rate boost for weaker sterilizers might be a good way to encourage this. Back-alley doctors pouring whiskey over a wound before going to work is a great mental image and something that's sadly very underused from what I've seen, even with access to actual sterilizine.

**Comment by JohnFulpWillard** [comment]
People who want to do surgeries should sign up to play Medical Doctor, not Station Engineer. Atmos techs can technically hack doors like Engineers despite the lack of their skillchip, yes, but wires are still related to Atmos tech's job. They have access to the SM and even have their own way of generating power through the turbine.

Compare this to Surgeries, which should solely be done by MDs/Robos, there's no reason for Paramedics/Chemists to be doing surgery because it's completely unrelated to their job. They don't have surgery access, don't spawn with any tools that lets them do surgery at all. Despite this, Paramedics often powergame their way to do MD's job regardless. It's helpful to the person being healed, maybe, sure. But it's incredibly boring for the Medical Doctor who hasn't had any patients in an hour or good for the game that people can just bypass the need for an entire job because everyone else does it for them.

On an unrelated note, it might be a good idea to make stuff like implanted surgery toolset implants give the trait? It's an augment so would be a nice way to encourage more players to get implanted (though not so strong for this, since it might be encouraging the wrong types of players to get said implant).

**Comment by JohnFulpWillard** [comment]
People who want to do surgeries should sign up to play Medical Doctor, not Station Engineer. Atmos techs can technically hack doors like Engineers despite the lack of their skillchip, yes, but wires are still related to Atmos tech's job. They have access to the SM and even have their own way of generating power through the turbine.

Compare this to Surgeries, which should solely be done by MDs/Robos, there's no reason for Paramedics/Chemists to be doing surgery because it's completely unrelated to their job. They don't have surgery access, don't spawn with any tools that lets them do surgery at all. Despite this, Paramedics often powergame their way to do MD's job regardless. It's helpful to the person being healed, maybe, sure. But it's incredibly boring for the Medical Doctor who hasn't had any patients in an hour or good for the game that people can just bypass the need for an entire job because everyone else does it for them.

On an unrelated note, it might be a good idea to make stuff like implanted surgery toolset implants give the trait? It's an augment so would be a nice way to encourage more players to get implanted (though not so strong for this, since it might be encouraging the wrong types of players to get said implant).

**Comment by tralezab** [comment]
Our game is run on an item economy, so I don't see why this is the one case we bypass said item economy entirely and just add debilitating passives to anyone who isn't the job. Instead, we could make surgery tools more lucrative, or harder for others to gain access to. Consider this part of your document:
> Surgical tools are easily obtained, to the point where it is simple for chemists or parameds to just waltz into med storage, print tools and instantly be able to play as a MD with better access to chem or other departments.

Based off of that, you're not going to investigate if you should be able to print every surgery tool for yourself with minimal effect on other players to the point where MDs just kinda let people do it? Why are we so worried about ruining ghetto surgery when the easy access anyone apparently has to real surgery tools is ruining ghetto surgery as we speak? Someone's gotta speak up for the fact that way too many items in this game are too easy to replenish at minimal cost to others. I mean, if it barely bothers medical that some resources got turned into surgery tools, from a **game** perspective they're not only barely inconvenienced for allowing it, but actually possibly benefit from their job being expanded to anyone who wants to join.

**Comment by Cheshify** [comment]
I really think the idea of "incentivizing doctors to do surgery" is the great route forward here and very smart people have put forward some great suggestions that I think would be looking into. As-is, this would screw over far more people than it would help doctors have a more solid niche, so granting a potential "surgery skillchip" an ability shouldn't be an ability everyone used to have.

**Comment by GuillaumePrata** [comment]
Time to make surgery tools need silver to be printed as this already is one of the resource bottlenecks of medbay.

**Comment by CRITAWAKETS** [comment]
I was going to pretty much state exactly what tralezab has said. Why not just make surgical items way harder to get? Right now you can just go to a protolathe or autolathe if you need any new tools. Or you know, just grab one of the 50 million surgical tools lying about the place in medical.

Solution's simple. Lock the basic items behind cargo which has more of a cost (either 15/20/30 minutes of your time or a lot of credits). As for the experimentals, it's a bit of a compromise because of just fluidity and such (alongside funny free golems) but it's probably best to keep them printable. Not having them be printable from autolathes already restricts their construction to medical and science.
Alien tools shouldnt recieve any changes, they're fine.

**Comment by maxspells** [comment]
I hear a lot of these same arguments, and I’d like to emphasize that this isn’t as much of a nerf as it’s being made out to be. Not every surgery step rolls a fail chance, and even if you do fail the steps that do, it just takes a few seconds to retry the step successfully. The base chance does not stack if the conditions were to make the fail% worse than the base, so this is only really going to be noticeable enough to be annoying to people who are trying to do surgery all of the time. There will still be demand for tools as better tools means faster surgery, which means you can retry a step you fail a lot quicker.

As for the item economy, I heard that there was already work being done on a lathe tax which I assume is going to tackle that issue as well.

Overall, I stand behind the changes here. The way it is implemented is a minor inconvenience to people wanting to sit and do surgery without the chip, while allowing players the option to still easily do it.

**Comment by CRITAWAKETS** [comment]
Oh whoops, noticed trale didn't actually state something important. Skillchips are kind of a bad system right now and i really wouldn't recommend putting anything serious behind it (wire readouts are already too much in my eyes). They're only obtainable from one location (head of staff locker) and you can only have one at a time, period, without a genetics augmentation. It's why it's better to focus on item economy for the tools instead of a skillchip, because skillchips just don't really work too well.

And if someone tells me skillchips are going to be fixed, engineering wire readouts literally give you ghetto all-access and that still hasn't convinced people to change skillchips.

**Comment by timothymtorres** [comment]
> Time to make surgery tools need silver to be printed as this already is one of the resource bottlenecks of medbay.

IRL a lot of surgery tools are made with silver for it's ability to kill bacteria and other microorganisms that touch it.  Making it require silver would somewhat restrict it slightly, which I think is fine.  Silver is one of the most common ores to find for miners.

**Comment by B4CKU** [comment]
dunno if anyone else suggested it, but i think that a skillchip that tells you the next step of the surgery as a balloon alert would be neat

**Comment by Mothblocks** [comment]
> Our game is run on an item economy, so I don't see why this is the one case we bypass said item economy entirely and just add debilitating passives to anyone who isn't the job. Instead, we could make surgery tools more lucrative, or harder for others to gain access to. Consider this part of your document:
> 
> > Surgical tools are easily obtained, to the point where it is simple for chemists or parameds to just waltz into med storage, print tools and instantly be able to play as a MD with better access to chem or other departments.
> 
> Based off of that, you're not going to investigate if you should be able to print every surgery tool for yourself with minimal effect on other players to the point where MDs just kinda let people do it? Why are we so worried about ruining ghetto surgery when the easy access anyone apparently has to real surgery tools is ruining ghetto surgery as we speak? Someone's gotta speak up for the fact that way too many items in this game are too easy to replenish at minimal cost to others. I mean, if it barely bothers medical that some resources got turned into surgery tools, from a **game** perspective they're not only barely inconvenienced for allowing it, but actually possibly benefit from their job being expanded to anyone who wants to join.

this hits it on the head IMHO

**Comment by zxaber** [comment]
If we're doing this, make the robo's only give the bonus for brain removal (organ manip on head, specifically). I really don't want people running up to robo because "you're one of the two jobs that can do <any surgery>, fix this person."

Like, I play roboticist to work on metal, not flesh. The brain removal is just the setup.

**Comment by maxspells** [comment]
>but yes skillchips are actually kind of a failed concept unless we took serious steps to gear /every/ job around the idea of being better inherently
> 
Out of all of the jobs that currently have skillchips, this would be one place where it makes sense to have it. MDs should be slightly better at doing surgery than other roles, and it makes sense that someone untrained would occasionally frick up a surgery step and have to try again. I opted doing a slight nerf over an MD buff because simply just buffing MDs isn't going to decentivize surgery parameds, or anyone else. The nerf is just minor enough to make people join as MD for the chip if they want to do surgery, and decentivizing being paramed and just printing off tools roundstart(lathe tax or material increase or not).

Players without the trait will still be able to do surgeries easily, and with the proper tools, this will only be noticeable enough to be annoying to people wanting to sit and do surgeries all the time without the chip. This also incentivizes ghetto surgery since the fail% was solely tied to a speed decrease that would bleed into fail chance if the tools and conditions were bad enough, but now there will always be a base chance to fail surgery no matter what tools they have. Players will be more likely to use ghetto surgery instead of just going out of their way to grab a set of surgery tools to bypass the downsides completely.

This will interact well with an item economy fix if there is one in the future like the lathe tax, as people will still want surgery tools but they will want them for faster surgeries instead of bypassing a fail%.

**Comment by ATH1909** [comment]
> If we're doing this, make the robo's only give the bonus for brain removal (organ manip on head, specifically). I really don't want people running up to robo because "you're one of the two jobs that can do [any surgery], fix this person."
> 
> Like, I play roboticist to work on metal, not flesh. The brain removal is just the setup.

aug surgery hello??

**Comment by GuillaumePrata** [comment]
> unless we took serious steps to gear /every/ job around the idea of being better inherently

Now is this a direction that has any support? I mostly hear people hating this idea.

I still think the chip (and this possible design direction) is an interesting concept because I'm tired of a random janitor dragging their metabuddy to medbay and fixing them himself because he doesn't trust doctors to do it properly and can revive his friend as efficiently, if not more because of metaknowledge, than the doctors on that shift.

**Comment by Ryll-Ryll** [comment]
I don't see any benefit to making it harder for normal crew members to perform surgery, and actually went through quite a bit of effort when designing Wounds and their treatments to offer accessibility to treatments to people not in a medical role and/or not in a fully stocked medbay. In almost a decade of time split between playing medbay, adminning, and coding, I have very rarely had the sense that it would be wise for the game or make it more enjoyable to nerf improvised medical treatments with the reason "it will make doctors more important".

Is there even really a problem with non-medical staff going through all the efforts to print off a complete set of surgical tools, setting up a treatment center in a public area to attract patients, and spending time actively treating people? Who are these assistants this PR is targeting who are abusing the fact that it's possible for them to do surgery (if they go through the time and effort to prep for it and are willing to set aside time to do it, in which case, they're probably just playing as a doctor)?

Surgery is already usually the treatment of last resort, and something I'd rather be made more accessible (like when removing embedded objects was changed from an entire surgery to just using a hemostat on the limb) than less accessible. Blocking this until convinced otherwise.

**Comment by Ryll-Ryll** [comment]
After talking with @Mothblocks we're gonna close it

**Pull request closed without merging**